src/python/pants/backend/scala/dependency_inference/symbol_mapper.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from collections import defaultdict
+from dataclasses import dataclass
+
+from pants.backend.scala.dependency_inference.scala_parser import ScalaSourceDependencyAnalysis
+from pants.backend.scala.target_types import ScalaSourceField
+from pants.build_graph.address import Address
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import AllTargets, Targets
+from pants.util.logging import LogLevel
+
+
+class SymbolMap:
```

Comment:
This could probably be merged with `PackageRootedDependencyMap` and then shared between Java and Scala backends. But that should be done separately as a cleanup.